src/services/agent-service.ts

```diff
@@ -94,15 +94,25 @@ export class AgentService {
     if (!this.snapshotService) { return response.json({success: false}) }
 
     const configuration = new ConfigurationService().configuration
+    // trim the string of whitespace and concat per-snapshot CSS with the globally specified CSS
+    const percySpecificCSS = configuration.snapshot['percy-css'].concat(request.body.percyCSS || '').trim() || ''
     const snapshotOptions: SnapshotOptions = {
+      percyCSS: percySpecificCSS,
       widths: request.body.widths || configuration.snapshot.widths,
       enableJavaScript: request.body.enableJavaScript != null
         ? request.body.enableJavaScript
         : configuration.snapshot['enable-javascript'],
       minHeight: request.body.minHeight || configuration.snapshot['min-height'],
     }
 
-    const domSnapshot = request.body.domSnapshot
+    let domSnapshot = request.body.domSnapshot
+    const percyCSSFileName = `${Date.now()}-percy-specific.css` as string
```

Comment:
Pedantic, but if this needs a timestamp I think `percy-specific.${Date.now()}.css` would be easier to find in the filesystem at a glance / search.

Reply:
I'm cool with it, matches the log files too